lib/index.js

```diff
@@ -73,7 +73,6 @@ const internals = {
 exports.wrap = function (error, statusCode, message) {
 
     Hoek.assert(error instanceof Error, 'Cannot wrap non-Error object');
-    Hoek.assert(!error.isBoom || (!statusCode && !message), 'Cannot provide statusCode or message with boom error');
```

Comment:
My understanding is that Boom has it for a guiding principle that Boom errors should not be re-wrapped. That's a fair principle, and it's been clearly exposed. Starting there, you can do one of two things when users try to call `Boom.wrap()` with a Boom object:
- throw an error if there are additional parameters (behavior introduced in [544cc13db7c7fc99b43bbb8eb1ce2ce57f2c78db](https://github.com/hapijs/boom/commit/544cc13db7c7fc99b43bbb8eb1ce2ce57f2c78db)). This is what I want to revert with the current PR on the grounds of backward compatibility and compliance with the documentation). What I also don't like with this solution is that the function has a less consistent behavior: it throws an error if other parameters are supplied, but returns the original object if not.
- return the original Boom object regardless of other parameters. If you do that, then `Boom.wrap()` can be used as a more friendly error wrapper when your code has to interact with inconsistent error types (sometimes Boom, sometimes not). This was the behavior until [544cc13db7c7fc99b43bbb8eb1ce2ce57f2c78db](https://github.com/hapijs/boom/commit/544cc13db7c7fc99b43bbb8eb1ce2ce57f2c78db).

I hope that clears up your questions.